runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/GradleDokkaSourceSetBuilder.kt

```diff
@@ -0,0 +1,218 @@
+@file:Suppress("FunctionName", "UnstableApiUsage")
+
+package org.jetbrains.dokka.gradle
+
+import com.android.build.gradle.api.AndroidSourceSet
+import groovy.lang.Closure
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.*
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.setProperty
+import org.gradle.util.ConfigureUtil
+import org.jetbrains.dokka.*
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import java.io.File
+import java.net.URL
+
+internal fun Task.GradleDokkaSourceSetBuilderFactory(): (name: String) -> GradleDokkaSourceSetBuilder =
+    { name -> GradleDokkaSourceSetBuilder(name, project) }
+
+
+open class GradleDokkaSourceSetBuilder constructor(
+    @Transient @get:Input val name: String,
+    @Transient @get:Internal internal val project: Project
+) : DokkaConfigurationBuilder<DokkaSourceSetImpl> {
+
+    @Internal
+    val sourceSetID: DokkaSourceSetID = DokkaSourceSetID(project, name)
+
+    @Classpath
+    @Optional
+    val classpath: ConfigurableFileCollection = project.files()
+
+    @Input
+    @Optional
+    val moduleDisplayName: Property<String?> = project.objects.safeProperty()
+
+    @Input
+    @Optional
+    val displayName: Property<String?> = project.objects.safeProperty()
```

Comment:
You should use Property<String> here. A property can always be _unset_ and usually that should be enough and there shouldn't be the need to set it to null. You can still use `@Optional` to state that _unset_ is a valid state.

Reply:
You mean I should use a Kotlin property? `var displayName: String? = null`

Reply:
Term _property_ is quite overloaded. :)

This is what I mean:

`val displayName: Property<String> = project.objects.property()`

or, if you can make the class abstract, just:

`abstract val displayName: Property<String>`

Reply:
Now I saw the other comment, which answers why you use `<String?>`. I'll answer there later.